Source/Nalu.Maui.Core/Platforms/iOS/MessageHandlerNSUrlSessionDownloadDelegate.cs

```diff
@@ -543,21 +579,18 @@ private NSMutableDictionary GetPlatformHeaders(HttpRequestMessage request, Cooki
             enumeratedHeaders.Select(object (h) => h.Key).ToArray()
         );
 
-        // set header cookies if needed from the managed cookie container if we do use Cookies
-        if (Session.Configuration.HttpCookieStorage is not null && cookieContainer is not null)
+        // Set header cookies if needed from the managed cookie container
+        if (cookieContainer is not null)
         {
-            // As per docs: An HTTP cookie header, with strings representing Cookie instances delimited by semicolons.
-            var cookies = cookieContainer.GetCookieHeader(request.RequestUri!);
-
-            if (!string.IsNullOrEmpty(cookies))
+            lock (cookieContainer)
```

Comment:
Not sure if you're aware of, but if not, maybe using the [`Lock`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.lock?view=net-10.0) type could bring some improvements here